python_modules/dagster/dagster/_core/asset_graph_view/asset_graph_view.py

```diff
@@ -362,56 +392,101 @@ def get_asset_slice_from_subset(self, subset: AssetSubset) -> Optional["AssetSli
         else:
             return None
 
-    def get_asset_slice_from_valid_subset(self, subset: ValidAssetSubset) -> "AssetSlice":
-        return _slice_from_valid_subset(self, subset)
+    def legacy_get_asset_slice_from_valid_subset(self, subset: ValidAssetSubset) -> "AssetSlice":
+        return AssetSlice(self, _AssetSliceCompatibleSubset(subset))
 
     def get_asset_slice_from_asset_partitions(
         self, asset_partitions: AbstractSet[AssetKeyPartitionKey]
     ) -> "AssetSlice":
         asset_keys = {akpk.asset_key for akpk in asset_partitions}
+        partition_keys = {
+            akpk.partition_key for akpk in asset_partitions if akpk.partition_key is not None
+        }
         check.invariant(len(asset_keys) == 1, "Must have exactly one asset key")
         asset_key = asset_keys.pop()
-        return _slice_from_valid_subset(
-            self,
-            ValidAssetSubset.from_asset_partitions_set(
-                asset_key=asset_key,
-                partitions_def=self._get_partitions_def(asset_key),
-                asset_partitions_set=asset_partitions,
-            ),
+        partitions_def = self._get_partitions_def(asset_key)
+        value = (
+            partitions_def.subset_with_partition_keys(partition_keys)
+            if partitions_def
+            else bool(asset_partitions)
+        )
+        return AssetSlice(
+            self, _AssetSliceCompatibleSubset(AssetSubset(key=asset_key, value=value))
         )
 
     @cached_method
     def create_empty_slice(self, *, asset_key: AssetKey) -> AssetSlice:
-        return _slice_from_valid_subset(
-            self,
-            AssetSubset.empty(asset_key, self._get_partitions_def(asset_key)),
+        partitions_def = self._get_partitions_def(asset_key)
+        value = partitions_def.empty_subset() if partitions_def else False
+        return AssetSlice(
+            self, _AssetSliceCompatibleSubset(AssetSubset(key=asset_key, value=value))
         )
 
     def compute_parent_asset_slice(
         self, parent_asset_key: AssetKey, asset_slice: AssetSlice
     ) -> AssetSlice:
-        return _slice_from_valid_subset(
```

Comment:
Here, I'm removing the `get_parent_asset_subset` method from `AssetGraph` (as this was the only call site), and moving it to be inside of the AssetGraphView. This allows us to get rid of some weird shuffling between subset and slice formats when we know we'll only ever want the slice format.